### PR TITLE
feat(journal): implement template feature

### PR DESCRIPTION
**Pull request by Gian12315**
This implements the templates feature for journal, the templates are local to the workspace used.

### CODE REVIEW

File: lua/neorg/modules/core/norg/journal/module.lua [hunk 1]
```diff
@@ -13,6 +13,9 @@ With this commands you can open the config files for the dates.
 - `Neorg journal custom`
 This command requires a date as an argument.
 The date should have to format yyyy-mm-dd.
+
+- `Neorg journal template`
```
Comment (max397574) [inline lua/neorg/modules/core/norg/journal/module.lua:17]: add `:` so it's the same as the other ones

File: lua/neorg/modules/core/norg/journal/module.lua [hunk 2]
```diff
@@ -60,10 +65,23 @@ module.private = {
...
+        end
+
         module.required["core.norg.dirman"].create_file(
             folder_name .. neorg.configuration.pathsep .. path,
             workspace or module.required["core.norg.dirman"].get_current_workspace()[1]
         )
+
+        if
+            not journal_file_exists
+            and use_template
```
Comment (max397574) [inline lua/neorg/modules/core/norg/journal/module.lua:80]: just use the value `module.config.public.use_template` here and not create a local variable

File: lua/neorg/modules/core/norg/journal/module.lua (hunk 1 above)
Comment (Gian12315) [reply lua/neorg/modules/core/norg/journal/module.lua]: Done, Neorg journal custom also needs a `:`

File: lua/neorg/modules/core/norg/journal/module.lua (hunk 2 above)
Comment (Gian12315) [reply lua/neorg/modules/core/norg/journal/module.lua]: Done

**Comment by vhyrro** [comment]
This is nice! Thanks for implementing it :)

**Pull request merged**